thread_pthread_mn.c

```diff
@@ -562,7 +567,12 @@ timer_thread_register_waiting(rb_thread_t *th, int fd, enum thread_sched_waiting
         flags |= thread_sched_waiting_timeout;
     }
 
+#if HAVE_SYS_EVENT_H
+    struct kevent ke[2];
+    int num_events = 0;
+#else
     uint32_t epoll_events = 0;
+#endif
     if (flags & thread_sched_waiting_timeout) {
         VM_ASSERT(rel != NULL);
         abs = rb_hrtime_add(rb_hrtime_now(), *rel);
```

Comment:
I originally had a EVFILT_TIMER defined here with `EV_SET` but it doesn't seem necessary? I haven't been able to see any odd behavior not having it in my testing so far.

Reply:
Based on the current MN implementation, there isn't really a good place to or need to use `EVFILT_TIMER`.